ccm_web/client/src/utils/handleErrors.ts

```diff
@@ -52,31 +59,31 @@ const handleErrors = async (resp: Response): Promise<void> => {
   if (resp.ok) return
   const text = await resp.text()
   console.error(text)
-  const errorBody: APIErrorData = JSON.parse(text)
+  let errorBody: APIErrorData
+  try {
+    errorBody = JSON.parse(text)
+  } catch (error) {
+    throw new Error(`Non-JSON error encountered with status code ${resp.status}`)
+  }
+
+  if (resp.status === 401 && errorBody.redirect === true) {
+    redirect('/')
+    throw new UnauthorizedError()
+  }
+  if (isCanvasAPIErrorData(errorBody)) throw new CanvasError(errorBody.errors)
 
+  const apiErrorMessage = Array.isArray(errorBody.message) ? errorBody.message.join(' ') : errorBody.message
   switch (resp.status) {
+    case 400:
+      throw new BadRequestError(apiErrorMessage)
     case 401:
-      if (errorBody.redirect === true) redirect('/')
       throw new UnauthorizedError()
```

Comment:
is there any reason why you moved this code above L69-72? I felt having that code here is clean.

Reply:
Hm, that might work. I don't quite remember why I did this.

Reply:
Okay, I moved it back: https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/340/commits/64c22681557910fa994d721dafc7e38419aa5f91